torchtune/modules/low_precision/_register_nf4_dispatch_ops.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torchao.dtypes.nf4tensor import implements as nf4_tensor_impl, to_nf4
+
+
+@nf4_tensor_impl([torch.ops.aten.clone.default])
+def clone(func, *args, **kwargs):
+    """
+    __torch_dispatch__ override that is called when cloning an NF4Tensor.
+    This is implemented by creating a new NF4Tensor with the unquantized weight
+    of the input tensor. Note that this is not an exact "clone" due to the loss
+    in precision.
+    """
+    return to_nf4(args[0][0].get_original_weight())
+
+
+@nf4_tensor_impl([torch.ops.aten.copy_.default])
+def inplace_copy(func, *args, **kwargs):
+    """
+    Performs an inplace copy of an incoming tensor into the tensor
+    being copied into. The inplace tensor is given by args[0][1] and the
+    tensor being copied into is given by args[0][0]. The copy is performed
+    by copying over all attributes. This method would have to be updated
+    if additional attributes are added to NF4Tensor.
```

Comment:
Can we do an assert on e.g. the attributes of `ref_tensor` to address this?

Reply:
This might be a little tricky as I don't want this CPU overhead everytime this op is called. We could add a unittest that alerts us to check these attributes if nf4 tensor changes?

Reply:
Added unittest for now.